use-cases/src/Contracts/CrowdFunding.hs

```diff
@@ -0,0 +1,119 @@
+{-# LANGUAGE TemplateHaskell #-}
+
+-- This is the fully parallel version that collects all contributions
+-- in a single transaction. This is, of course, limited by the maximum
+-- number of inputs a transaction can have.
+
+module Contracts.CrowdFunding (contribute, collect, refund)
+where
+
+import           Control.Monad (void)
+import           Plutus
+{-# ANN module "HLint: ignore Avoid restricted function" #-}
+
+
+-- Functionality for campaign contributors
+-- --
+
+-- Contribute funds to the campaign (contributor)
+--
+contribute :: PubKey -> Value -> T TxOutRef
+contribute ownerPubKey value
+  = do
+    { assert (value > 0)
+    ; contributorPubKey <- lookupMyPubKey
+    ; myPayment         <- createPayment (value + standardTxFee)
+    ; submitTransaction Tx
+      { txInputs  = [myPayment]
+      , txOutputs = [TxOutScript
+                       value
+                       (contributionScript ownerPubKey)
+                       [| contributorPubKey |] -- data script: ought to be lifted into PLC at coordination runtime
+                    ]
+      }
+    }

```

Comment:
Do not know what others think, but I'd prefer to keep indentation-sensitive `do` as everywhere else in the development and save the `{,,}` syntax for records.